hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -233,6 +235,23 @@ public HoodieTableConfig(FileSystem fs, String metaPath, String payloadClassName
         "hoodie.properties file seems invalid. Please check for left over `.updated` files if any, manually copy it to hoodie.properties and retry");
   }
 
+  private static Properties getOrderedPropertiesWithTableChecksum(Properties props) {
+    LinkedHashMap<String, String> propsMap = getOrderedPropertiesMap(props);
+    propsMap.put(TABLE_CHECKSUM.key(), String.valueOf(generateChecksum(props)));
+    Properties orderedProps = new Properties();
```

Comment:
This will destroy the ordering, right? You need to control the iteration during `store`, you to create class that is similar to how this was handled in `TypedProperties`

```
class OrderedProperties extends Properties {
  LinkedHashSet hashSet;
  
  Enumeration<> keys() { hashSet.keys() }
}
```

Reply:
+1 internally this is just a hashtable right. so ordering is not guaranteed.

Reply:
yeah sorry, this doesn't guarantee order. i just wanted to quickly revert TypedProperties change and run long-running integ tests to verify that CME does not happen. I'll udpate the PR with ordering. This ordering will only be done for HoodieTableConfig just before store.

Reply:
sure, makes sense.